io.quarkus.runtime.configuration.QuarkusConfigFactory not a subtype in v3.0.0.Alpha3 dev mode

### Describe the bug

When you define a ConfigProperty and forget to set an actual value you get a very convenient error message (when running in Dev Mode) in the UI that shows a form where you can set the value for this missing property.  In the latest v3.0.0.Alpha3 version this appears to be broken.  

The form shows up, but when I enter a value and submit, I get an Internal Server Error served at http://localhost:8080/io.quarkus.vertx-http.devmode.config.fix

The form (note, this is not the bug!)
![Screenshot from 2023-01-23 12-32-50](https://user-images.githubusercontent.com/373537/214030442-3e1c8bfa-1187-47ef-9ac8-87fb2a9eed0b.png)




### Expected behavior

The value of the property is set in your application.properties file

### Actual behavior
![image](https://user-images.githubusercontent.com/373537/214030620-b1562c1b-6361-404b-bcdf-5118b0d3b0e5.png)


2023-01-23 12:32:54,124 ERROR [io.ver.ext.web.RoutingContext] (vert.x-eventloop-thread-1) Unhandled exception in router: java.lang.RuntimeException: java.util.ServiceConfigurationError: io.smallrye.config.SmallRyeConfigFactory: io.quarkus.runtime.configuration.QuarkusConfigFactory not a subtype
        at io.quarkus.vertx.http.deployment.devmode.console.ConfigEditorProcessor.updateConfig(ConfigEditorProcessor.java:258)
        at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$3$1.handle(VertxHttpHotReplacementSetup.java:115)
        at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$3$1.handle(VertxHttpHotReplacementSetup.java:101)
        at io.vertx.core.impl.ContextBase.lambda$null$0(ContextBase.java:137)
        at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:264)
        at io.vertx.core.impl.ContextBase.lambda$executeBlocking$1(ContextBase.java:135)
        at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.util.ServiceConfigurationError: io.smallrye.config.SmallRyeConfigFactory: io.quarkus.runtime.configuration.QuarkusConfigFactory not a subtype
        at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:593)
        at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1244)
        at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1273)
        at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1309)
        at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1393)
        at io.smallrye.config.SmallRyeConfigProviderResolver.getFactoryFor(SmallRyeConfigProviderResolver.java:100)
        at io.smallrye.config.SmallRyeConfigProviderResolver.getConfig(SmallRyeConfigProviderResolver.java:76)
        at io.smallrye.config.SmallRyeConfigProviderResolver.getConfig(SmallRyeConfigProviderResolver.java:64)
        at org.eclipse.microprofile.config.ConfigProvider.getConfig(ConfigProvider.java:85)
        at io.quarkus.runtime.configuration.ConfigUtils.getProfiles(ConfigUtils.java:273)
        at io.quarkus.vertx.http.deployment.devmode.console.ConfigEditorProcessor.updateConfig(ConfigEditorProcessor.java:218)
        ... 13 more



### How to Reproduce?

1. Create new project: `quarkus create app com.redhat.developers:mytest -P3.0.0.Alpha3`

2. Edit the supplied GreetingResource.java and add 
```    
   @ConfigProperty(name="greeting")
    String greeting;
```
3. Start dev mode `quarkus dev`
4. In your browser, go to localhost:8080/hello
5. Notice the error with the missing property value and the form field to add a value.  Populate with any string and submit
6. You should see an "Internal Server Error"


### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk version "17.0.5" 2022-10-18 OpenJDK Runtime Environment (Red_Hat-17.0.5.0.8-1.fc37) (build 17.0.5+8) OpenJDK 64-Bit Server VM (Red_Hat-17.0.5.0.8-1.fc37) (build 17.0.5+8, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.0.0.Alpha3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/ConfigEditorProcessor.java
+++ b/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/ConfigEditorProcessor.java
@@ -1,7 +1,5 @@
 package io.quarkus.vertx.http.deployment.devmode.console;
 
-import static io.quarkus.runtime.LaunchMode.DEVELOPMENT;
-
 import java.io.BufferedWriter;
 import java.io.IOException;
 import java.nio.file.Files;
@@ -46,7 +44,6 @@ import io.quarkus.dev.console.DevConsoleManager;
 import io.quarkus.devconsole.runtime.spi.DevConsolePostHandler;
 import io.quarkus.devconsole.spi.DevConsoleRouteBuildItem;
 import io.quarkus.devconsole.spi.DevConsoleRuntimeTemplateInfoBuildItem;
-import io.quarkus.runtime.configuration.ConfigUtils;
 import io.quarkus.vertx.http.runtime.devmode.ConfigDescription;
 import io.quarkus.vertx.http.runtime.devmode.ConfigDescriptionsManager;
 import io.quarkus.vertx.http.runtime.devmode.ConfigDescriptionsRecorder;
@@ -215,7 +212,6 @@ public class ConfigEditorProcessor {
         if (values != null && !values.isEmpty()) {
             try {
                 Path configPath = getConfigPath();
-                List<String> profiles = ConfigUtils.getProfiles();
                 List<String> lines = Files.readAllLines(configPath);
                 for (Map.Entry<String, String> entry : values.entrySet()) {
                     String name = entry.getKey();
@@ -223,17 +219,11 @@ public class ConfigEditorProcessor {
                     int nameLine = -1;
                     for (int i = 0, linesSize = lines.size(); i < linesSize; i++) {
                         String line = lines.get(i);
-                        for (String profile : profiles) {
-                            String profileName = !profile.equals(DEVELOPMENT.getDefaultProfile()) ? "%" + profile + "." + name
-                                    : name;
-                            if (line.startsWith(profileName + "=")) {
-                                name = profileName;
-                                nameLine = i;
-                                break;
-                            }
+                        if (line.startsWith(name + "=")) {
+                            nameLine = i;
+                            break;
                         }
                     }
-
                     if (nameLine != -1) {
                         if (value.isEmpty()) {
                             lines.remove(nameLine);

